### PR TITLE
Feature/api validation

### PR DESCRIPTION
**Pull request by zakhenry**

### CODE REVIEW

**Comment by ghost** [comment]
Pull request looks fine. Waiting for conflicts to be resolved and tests to pass.

File: api/app/Http/Controllers/ChildEntityController.php [hunk 1]
```diff
@@ -122,17 +122,17 @@ public function postOne($id, Request $request)
...
-        if ($childModel->exists) {
-            $validationRules = $this->addIdOverrideValidationRule($validationRules, $childId);
-        }
-        $this->validateRequest($request->all(), $validationRules, $childModel->exists);
+        $parent = $this->findParentEntity($id);
+
+        $this->checkEntityIdMatchesRoute($request, $childId, $this->getChildModel());
+        $childModel = $this->findOrNewChildEntity($childId, $parent);
+
+        $this->validateRequest($request->all(), $this->addIdOverrideValidationRule($this->getValidationRules(), $childId));
```
Comment (Redjik) [inline api/app/Http/Controllers/ChildEntityController.php:132]: it's either `heckEntityIdMatchesRoute` or `addIdOverrideValidationRule`

File: api/src/Model/Model/BaseModel.php [hunk 2]
```diff
@@ -392,4 +392,35 @@ protected function getRelationCacheKey($method)
...
+    public function hydrateRequestCollection(array $requestCollection, Collection $existingModels = null)
+    {
+        $keyName = $this->getKeyName();
+        $models = array_map(function ($item) use ($keyName, $existingModels) {
+
+            $model = null;
+            $entityId = $item[$keyName];
+
+            if ($existingModels) {
+                $model = $existingModels->get($entityId, $this->newInstance());
```
Comment (Redjik) [inline api/src/Model/Model/BaseModel.php:414]: new Instance is created on each iteration

File: api/app/Http/Controllers/ChildEntityController.php (hunk 1 above)
Comment (Redjik) [inline api/app/Http/Controllers/ChildEntityController.php:132]: the same for other 3 places, to sum up it should be done for
`EntityController` and `ChildEntityController`
`putOne`, `patchOne`

File: api/src/Model/Model/BaseModel.php (hunk 2 above)
Comment (zakhenry) [inline api/src/Model/Model/BaseModel.php:414]: Ah, good point. I'll fix that up.

File: api/app/Http/Controllers/ChildEntityController.php (hunk 1 above)
Comment (zakhenry) [inline api/app/Http/Controllers/ChildEntityController.php:132]: @Redjik yep I implemented it for those other cases.

**Pull request merged**